ntoskrnl/ke/thrdobj.c

```diff
@@ -1137,8 +1139,8 @@ KeSetSystemAffinityThread(IN KAFFINITY Affinity)
         }
 
         /* Calculate the ideal CPU from the affinity set */
-        BitScanReverse(&NodeMask, AffinitySet);
-        CurrentThread->IdealProcessor = (UCHAR)NodeMask;
+        BitScanReverseAffinity(&IdealProcessor, AffinitySet);
+        CurrentThread->IdealProcessor = (UCHAR)IdealProcessor;
```

Comment:
Interesting, that the cast is required. I wonder why they used a UCHAR in the TEB (and then of course they upgraded it to a ULONG in windows 7 or 8)

Reply:
probably ^^